### PR TITLE
Add python 3.11 to the CI

### PR DESCRIPTION
**Pull request by bdraco**

### CODE REVIEW

**Comment by bdraco** [comment]
blocked until HA updates via https://github.com/home-assistant/core/pull/78860

File: pyproject.toml [hunk 1]
```diff
@@ -30,6 +30,7 @@ bleak-retry-connector = ">=2.3.0"
 orjson = ">=3.7.8"
 async-timeout = "^4.0.2"
 chacha20poly1305 = "^0.0.3"
+aiohttp = ">=3.8.3"
```
Comment (Jc2k) [inline pyproject.toml:33]: I thought this was just a dev dependency (to get asyncio support in pytest) - do we use it for something else now?

Comment (bdraco) [reply pyproject.toml:33]: It should just be dev.

Comment (bdraco) [reply pyproject.toml]: fixed. good catch

**Review by Jc2k — APPROVED** [review]

**Comment by bdraco** [comment]
Will push a tag in a moment which should take care of the coap fix as well

**Pull request merged**